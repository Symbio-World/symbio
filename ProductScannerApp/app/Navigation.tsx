import * as React from 'react'

import { createStackNavigator } from '@react-navigation/stack'

import {
  cardStackScreenOptions,
  modalCardStackScreenOptions,
} from './NavigationConfig'
import { Home } from './Home'
import { Test } from './modals/Test'
import { Test1 } from './modals/Test1'

export type RootStackParamList = {
  Main: {}
  Modals: {}
}

export type MainModalStackParamList = {
  Home: {}
}

export type RootModalStackParamList = {
  Test: { modalsInStack: number }
  Test1: { modalsInStack: number }
}

const RootStack = createStackNavigator<RootStackParamList>()
const MainStack = createStackNavigator<MainModalStackParamList>()
const ModalStack = createStackNavigator<RootModalStackParamList>()

const ModalStackComponent = () => (
  <ModalStack.Navigator headerMode="none" {...modalCardStackScreenOptions}>
    <ModalStack.Screen
      name="Test"
      component={Test}
      initialParams={{ modalsInStack: 0 }}
      options={{
        gestureEnabled: false,
        headerShown: false,
        cardShadowEnabled: true,
      }}
    />
    <ModalStack.Screen
      name="Test1"
      component={Test1}
      initialParams={{ modalsInStack: 0 }}
      options={{
        gestureEnabled: false,
        headerShown: false,
        cardShadowEnabled: true,
      }}
    />
  </ModalStack.Navigator>
)

const MainStackComponent = () => (
  <MainStack.Navigator initialRouteName="Home" headerMode="none">
    <MainStack.Screen name="Home" component={Home} />
  </MainStack.Navigator>
)

export const RootStackView = () => {
  return (
    <RootStack.Navigator
      headerMode="none"
      initialRouteName="Main"
      {...cardStackScreenOptions}
    >
      <RootStack.Screen name="Main" component={MainStackComponent} />
      <RootStack.Screen name="Modals" component={ModalStackComponent} />
    </RootStack.Navigator>
  )
}
