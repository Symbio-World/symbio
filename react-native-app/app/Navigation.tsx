import * as React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { ScanBarcodeScreen } from './barcode'
import { ProductViewScreen } from './product'

export type RootStackParamList = {
  ScanBarcodeScreen: {}
  ProductScreen: { barcode: string }
  FeedbackScreen: { title: string }
}

const RootStack = createStackNavigator<RootStackParamList>()

export const Navigation = () => {
  return (
    <NavigationContainer>
      <RootStack.Navigator initialRouteName="ScanBarcodeScreen">
        <RootStack.Screen
          name="ScanBarcodeScreen"
          component={ScanBarcodeScreen}
        />
        <RootStack.Screen
          name="ProductViewScreen"
          component={ProductViewScreen}
          options={{
            headerShown: true,
          }}
        />
      </RootStack.Navigator>
    </NavigationContainer>
  )
}
