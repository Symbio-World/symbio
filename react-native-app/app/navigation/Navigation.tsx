import * as React from 'react'
import {
  NavigationContainer,
  ParamListBase,
  NavigationContainerRef,
} from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { ScanBarcodeScreen } from '../barcode'
import { ProductScreen } from '../product'
import { GetUserEmailScreen } from '../get-user-email'
import { SetupPrinciplesScreen } from '../principle'
import {
  cardStackScreenOptions,
  modalCardStackScreenOptions,
} from './NavigationConfig'
import { createSwipableModal } from './createSwipableModal'

type SubNavigator<T extends ParamListBase> = {
  [K in keyof T]: { screen?: K; params?: T[K] }
}[keyof T]

export type RootStackParamList = {
  Main: {}
  Modals: {}
}

export type MainStackParamList = {
  ScanBarcodeScreen: {}
  Modals: SubNavigator<ModalStackParamList>
}

export type ModalStackParamList = {
  ProductScreen: { barcode: string }
  GetUserEmailScreen: {}
  SetupPrinciplesScreen: {}
  Main: SubNavigator<MainStackParamList>
}

const RootStack = createStackNavigator<RootStackParamList>()
const MainStack = createStackNavigator<MainStackParamList>()
const ModalStack = createStackNavigator<ModalStackParamList>()

const ModalStackComponent = () => {
  return (
    <ModalStack.Navigator
      mode="modal"
      headerMode="none"
      {...modalCardStackScreenOptions}
    >
      <ModalStack.Screen
        name="ProductScreen"
        component={createSwipableModal(ProductScreen)}
        options={{
          gestureEnabled: false,
          headerShown: false,
          cardShadowEnabled: true,
        }}
      />
      <ModalStack.Screen
        name="GetUserEmailScreen"
        component={createSwipableModal(GetUserEmailScreen)}
        options={{
          gestureEnabled: false,
          headerShown: false,
          cardShadowEnabled: true,
        }}
      />
      <ModalStack.Screen
        name="SetupPrinciplesScreen"
        component={createSwipableModal(SetupPrinciplesScreen)}
        options={{
          gestureEnabled: false,
          headerShown: false,
          cardShadowEnabled: true,
        }}
      />
    </ModalStack.Navigator>
  )
}

const MainStackComponent = () => {
  return (
    <MainStack.Navigator headerMode="none" initialRouteName="ScanBarcodeScreen">
      <MainStack.Screen
        name="ScanBarcodeScreen"
        component={ScanBarcodeScreen}
      />
    </MainStack.Navigator>
  )
}

const navigationRef = React.createRef<NavigationContainerRef>()

export const navigate = <RouteName extends keyof RootStackParamList>(
  name: RouteName,
  args: RootStackParamList[RouteName],
) => {
  navigationRef.current?.navigate(name, args)
}

export const Navigation = () => {
  return (
    <NavigationContainer ref={navigationRef}>
      <RootStack.Navigator
        headerMode="none"
        mode="modal"
        initialRouteName="Main"
        {...cardStackScreenOptions}
      >
        <RootStack.Screen
          name="Main"
          component={MainStackComponent}
          options={{
            title: '',
          }}
        />
        <RootStack.Screen
          name="Modals"
          component={ModalStackComponent}
          options={{ headerShown: false }}
        />
      </RootStack.Navigator>
    </NavigationContainer>
  )
}
