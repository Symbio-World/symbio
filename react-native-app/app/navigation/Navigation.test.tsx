import * as React from 'react'
import { render } from 'react-native-testing-library'

import { Navigation } from './Navigation'

jest.mock('react-native-reanimated', () =>
  require('react-native-reanimated/mock'),
)

describe('Navigation', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Navigation />)
    expect(toJSON()).toMatchSnapshot()
  })
})
