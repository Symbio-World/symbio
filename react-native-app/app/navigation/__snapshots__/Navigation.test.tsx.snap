// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgba(0,0,0,0)",
                        },
                      ],
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        onLayout={[Function]}
                        style={
                          Object {
                            "flex": 1,
                            "overflow": "hidden",
                          }
                        }
                      >
                        <View
                          pointerEvents="box-none"
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <View
                            accessibilityElementsHidden={false}
                            closing={false}
                            gestureVelocityImpact={0.3}
                            importantForAccessibility="auto"
                            onClose={[Function]}
                            onGestureBegin={[Function]}
                            onGestureCanceled={[Function]}
                            onOpen={[Function]}
                            onTransitionStart={[Function]}
                            pointerEvents="box-none"
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                            transitionSpec={
                              Object {
                                "close": Object {
                                  "animation": "spring",
                                  "config": Object {
                                    "damping": 500,
                                    "mass": 3,
                                    "overshootClamping": true,
                                    "restDisplacementThreshold": 0.01,
                                    "restSpeedThreshold": 0.01,
                                    "stiffness": 1000,
                                  },
                                },
                                "open": Object {
                                  "animation": "spring",
                                  "config": Object {
                                    "damping": 500,
                                    "mass": 3,
                                    "overshootClamping": true,
                                    "restDisplacementThreshold": 0.01,
                                    "restSpeedThreshold": 0.01,
                                    "stiffness": 1000,
                                  },
                                },
                              }
                            }
                          >
                            <View
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "flex": 1,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                onGestureHandlerEvent={[Function]}
                                onGestureHandlerStateChange={[Function]}
                                style={
                                  Object {
                                    "flex": 1,
                                    "transform": Array [
                                      Object {
                                        "translateX": 0,
                                      },
                                      Object {
                                        "translateX": 0,
                                      },
                                    ],
                                  }
                                }
                              >
                                <View
                                  pointerEvents="none"
                                  style={
                                    Object {
                                      "backgroundColor": "rgb(242, 242, 242)",
                                      "bottom": 0,
                                      "left": 0,
                                      "position": "absolute",
                                      "shadowColor": "#000",
                                      "shadowOffset": Object {
                                        "height": 1,
                                        "width": -1,
                                      },
                                      "shadowOpacity": 0.3,
                                      "shadowRadius": 5,
                                      "top": 0,
                                      "width": 3,
                                    }
                                  }
                                />
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "flex": 1,
                                        "overflow": "hidden",
                                      },
                                      Array [
                                        Object {
                                          "backgroundColor": "rgb(242, 242, 242)",
                                        },
                                        undefined,
                                      ],
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "flexDirection": "column-reverse",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "flex": 1,
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Array [
                                            Object {
                                              "flex": 1,
                                            },
                                          ]
                                        }
                                      >
                                        <BarcodePicker
                                          onBarcodeFrameAvailable={[Function]}
                                          onChangeTrackedCodes={[Function]}
                                          onPropertyChanged={[Function]}
                                          onRecognizeNewCodes={[Function]}
                                          onScan={[Function]}
                                          onSettingsApplied={[Function]}
                                          onTextRecognized={[Function]}
                                          onWarnings={[Function]}
                                          scanSettings={
                                            ScanSettings {
                                              "symbologies": Object {
                                                "code128": SymbologySettings {
                                                  "checksums": Array [],
                                                  "enabled": true,
                                                  "extensions": Array [],
                                                },
                                                "code39": SymbologySettings {
                                                  "activeSymbolCounts": Array [
                                                    7,
                                                    8,
                                                    9,
                                                    10,
                                                    11,
                                                    12,
                                                    13,
                                                    14,
                                                    15,
                                                    16,
                                                    17,
                                                    18,
                                                    19,
                                                    20,
                                                  ],
                                                  "checksums": Array [],
                                                  "enabled": true,
                                                  "extensions": Array [],
                                                },
                                                "data-matrix": SymbologySettings {
                                                  "checksums": Array [],
                                                  "enabled": true,
                                                  "extensions": Array [],
                                                },
                                                "ean13": SymbologySettings {
                                                  "checksums": Array [],
                                                  "enabled": true,
                                                  "extensions": Array [],
                                                },
                                                "ean8": SymbologySettings {
                                                  "checksums": Array [],
                                                  "enabled": true,
                                                  "extensions": Array [],
                                                },
                                                "itf": SymbologySettings {
                                                  "checksums": Array [],
                                                  "enabled": true,
                                                  "extensions": Array [],
                                                },
                                                "qr": SymbologySettings {
                                                  "checksums": Array [],
                                                  "enabled": true,
                                                  "extensions": Array [],
                                                },
                                                "upca": SymbologySettings {
                                                  "checksums": Array [],
                                                  "enabled": true,
                                                  "extensions": Array [],
                                                },
                                                "upce": SymbologySettings {
                                                  "checksums": Array [],
                                                  "enabled": true,
                                                  "extensions": Array [],
                                                },
                                              },
                                            }
                                          }
                                          style={
                                            Array [
                                              Object {
                                                "flex": 1,
                                              },
                                            ]
                                          }
                                        />
                                        <View
                                          style={
                                            Array [
                                              Object {
                                                "position": "absolute",
                                              },
                                              Object {
                                                "bottom": 0,
                                              },
                                              Object {
                                                "width": "100%",
                                              },
                                              Object {
                                                "height": 64,
                                              },
                                              Object {
                                                "justifyContent": "center",
                                              },
                                              Object {
                                                "backgroundColor": "#000",
                                              },
                                              Object {
                                                "opacity": 0.5,
                                              },
                                            ]
                                          }
                                        >
                                          <Text
                                            style={
                                              Array [
                                                Object {
                                                  "color": "#fff",
                                                },
                                                Object {
                                                  "textAlign": "center",
                                                },
                                                Object {
                                                  "fontWeight": "bold",
                                                },
                                                Object {
                                                  "fontSize": 30,
                                                },
                                              ]
                                            }
                                          >
                                            Scan barcode
                                          </Text>
                                        </View>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
